src/components/GeneralModal/index.jsx

```diff
@@ -0,0 +1,243 @@
+import React, { useState } from 'react';
+
+import Box from '@mui/material/Box';
+import Typography from '@mui/material/Typography';
+import Modal from '@mui/material/Modal';
+import TextField from '@mui/material/TextField';
+
+import './generalModal.scss';
+
+const style = {
+  position: 'absolute',
+  top: '50%',
+  left: '50%',
+  transform: 'translate(-50%, -50%)',
+  width: 400,
+  bgcolor: 'background.paper',
+  border: '2px solid #000',
+  boxShadow: 24,
+  p: 4,
+};
+
+export default function BasicModal(props) {
+  const { isActive, setIsActive } = props;
+
+  const handleOpen = () => setIsActive(true);
+
+  const handleClose = () => {
+    setIsActive(false);
+    setFormType(true);
+  }
+
+  const [formType, setFormType] = useState(true);
+
+  const [email, setEmail] = useState('');
+
+  const [firstName, setFirstName] = useState('');
+
+  const [lastName, setLastName] = useState('');
+
+  const [phone, setPhone] = useState('');
+
+  const [password, setPassword] = useState('');
+
+  const [repeatPassword, setRepeatPassword] = useState('');
+
+  const [formError, setFormError] = useState('');
+
+  const [isUserAuthorized, setIsUserAuthorized] = useState(false);
+
+  const handleSubmit = () => {
+    if((password === repeatPassword) && (password ?? repeatPassword)) {
```

Comment:
```suggestion
    if((password === repeatPassword) && (password ?? repeatPassword)) {
```

i don't really get why do you use ?? here?

Reply:
fixed